nemo/core/classes/modelPT.py

```diff
@@ -155,3 +179,15 @@ def setup_optimization(self, optim_config: Optional[Union[DictConfig, Dict]] = N
             logging.info("Optimizer config = %s", str(optimizer))
 
             return optimizer
+
+    def configure_optimizers(self):
+        if self._scheduler is None:
+            return self._optimizer
+        else:
+            return [self._optimizer], [self._scheduler]
+
+    def train_dataloader(self):
+        return self._train_dl
+
+    def val_dataloader(self):
```

Comment:
Maybe setup `test_dataloader(self)` as well?

Reply:
added

Reply:
actually there is some PTL strangeness with this, so not doing it for now

Reply:
Sounds good.